Add an RFD spell check step to the lint workflow

rosstimothy opened the pull request:
In order for the action to pass various RFDs were updated to fix misspellings, and the dictionary was expanded.

fheinecke commented on .github/workflows/lint.yaml:
```diff
@@ -26,6 +27,8 @@ jobs:
           ref: ${{ github.event.pull_request.head.ref || github.event.merge_group.head_ref }}
           filters: |
+            has_rfd:
+              - 'rfd/**'
```
```suggestion
              - 'rfd/**.md'
```

This pattern should exclude `cspell.json` and `rfd/assets/**` changes

fheinecke replied:
```suggestion
              - 'rfd/**'
              - '.github/workflows/lint.yaml'
```

This workflow should be included as well so that changes to the workflow are validated at PR time

fheinecke commented on .github/workflows/lint.yaml:
```diff
@@ -231,3 +234,25 @@ jobs:
+    name: Lint (RFD)
+    needs: changes
+    if: ${{ !startsWith(github.head_ref, 'dependabot/') && needs.changes.outputs.has_rfd == 'true' }}
+    runs-on: ubuntu-22.04-16core
```
Is a 16 core runner needed?

fheinecke commented on .github/workflows/lint.yaml:
```diff
@@ -231,3 +234,25 @@ jobs:
+  lint-rfd:
+    name: Lint (RFD)
+    needs: changes
+    if: ${{ !startsWith(github.head_ref, 'dependabot/') && needs.changes.outputs.has_rfd == 'true' }}
```
Why exclude dependabot PRs?

fheinecke replied:
It should be fine to leave for consistency, but it shouldn't really be needed as dependabot should never change anything in this directory.

fheinecke commented on .github/workflows/lint.yaml:
```diff
@@ -231,3 +234,25 @@ jobs:
+      contents: read
+
+    container:
+      image: ghcr.io/gravitational/teleport-buildbox:teleport17
```
What tools are used from the buildbox? AFAIK we are not currently pinning pnpm or corepack in the buildbox (IIRC this was part of yesterday's UID/GID issue), so i'm not sure that we need to pull in this massive image on every commit.

fheinecke commented on .github/workflows/lint.yaml:
```diff
@@ -231,3 +234,25 @@ jobs:
+        uses: actions/checkout@v4
+
+      - name: Install JS dependencies
+        run: pnpm install --frozen-lockfile
```
Do we need all deps or just cspell?

fheinecke commented on rfd/cspell.json:
```diff
@@ -3,98 +3,322 @@
+    "APIMFA",
+    "APNG",
+    "AQAB",
+    "ASDFGHJK",
```
cSpell supports comments to allow specific works on specific files. For example, with yaml, you can write:

```yaml
# cspell:words someContextSpecificWordOrAcronym
```

Do all of these need to be added globally, or should some be allowed locally?

fheinecke replied:
Ah fair enough - I didn't catch that part.

fheinecke commented on .github/workflows/lint.yaml:
```diff
@@ -231,3 +236,25 @@ jobs:
+        run: pnpm install --frozen-lockfile
+
+      - name: Check spelling
+        run: pnpm cspell -c ./rfd/cspell.json rfd/**.md
```
nit: this should probably have `shopt -s nullglob` set

(sorry, missed this on first pass)

```suggestion
        run: shopt -s nullglob && pnpm cspell -c ./rfd/cspell.json rfd/**.md
```

```suggestion
        run: |
          shopt -s nullglob
          pnpm cspell -c ./rfd/cspell.json rfd/**.md
```

rosstimothy committed:
Add an RFD spell check step to the lint workflow

In order for the action to pass various RFDs were updated to fix
misspellings, and the dictionary was expanded.

fheinecke reviewed (approved):
Missed something minor on first pass. Eitehr way, LGTM
